challenge_01/base_solution.py

```diff
@@ -0,0 +1,103 @@
+from typing import List, Tuple
+
+import torch
+import torch.nn as nn
+import torch.utils.data as data
+
+
+class MultiLayerPerceptron(nn.Module):
+    """
+    Model class follow rules by nn.Module
+    I tried to fill class by glues I got.
+    Key trick is create hidden layers by blocks follow nn.Sequential
+    """
+
+    def __init__(
+        self,
+        layer_dim_list: List[int],
+        activation: nn.modules.activation,
+        batch_norm: bool,
+        dropout_p: float,
+    ) -> None:
+        super().__init__()
+        self.layer_dim_list = layer_dim_list
+        self.activation = activation
+        self.batch_norm = batch_norm
+        self.dropout_p = dropout_p
+        self._create_mlp()
+
+    def _create_block(self, n_in: int, n_out: int, batch_norm: bool) -> nn.Module:
+        if batch_norm:
+            block = nn.Sequential(
+                nn.Linear(n_in, n_out),
+                nn.BatchNorm1d(n_out),
+                nn.Dropout(p=self.dropout_p),
+                self.activation(),
+            )
+        else:
+            block = nn.Sequential(
+                nn.Linear(n_in, n_out), nn.Dropout(p=self.dropout_p), self.activation()
+            )
+        return block
+
+    def _create_mlp(self):
+        blocks = []
+        blocks.append(nn.Flatten())
+        for i in range(len(self.layer_dim_list) - 2):
+            blocks.append(
+                self._create_block(
+                    self.layer_dim_list[i], self.layer_dim_list[i + 1], self.batch_norm
+                )
+            )
+        blocks.append(nn.Linear(self.layer_dim_list[-2], self.layer_dim_list[-1]))
+        self.layer = nn.Sequential(*blocks)
```

Comment:
lepší je kdyby to vracelo layer a zapíšeš to v init (to mě naučil jiřík)

Reply:
ked to hovorí jiřík tak je to pravda!